Draft: Feature/5 add user

PreafericitulMarcus opened the pull request:
Done #5.

Review: not hard to do.

Time taken: 15 minutes.

Future improvement: validation for input variables.

PreafericitulMarcus committed:
feat: add new user with Full Name, return user UUID.

PreafericitulMarcus committed:
Merge branch 'master' into feature/5_add_user

Taveeh commented on backend/src/main/java/clf/integra/backend/service/UserService.java:
```diff
@@ -1,12 +1,31 @@
+        this.userRepository = userRepository;
+    }
+
+    /* Input: Full name of the User. Output: UUID of the newly added User*/
```
If you want to use javadocs, you should use them ike /** ... */, multiple line comment

Taveeh commented on backend/src/main/java/clf/integra/backend/controller/TestController.java:
```diff
@@ -25,6 +27,7 @@ public String test() {
     //Testing creation of the repository
     @GetMapping("/test-repo")
     public String testRepo() {
+        userService.addUserByName("A", "B", "C");
```
this is not necessary here, it was just to test the first ticket, will be removed soon

Taveeh reviewed (commented):
The problem here is that it does not meet the acceptance criteria -> no endpoint to add user

Taveeh reviewed (changes requested):
Create a user repo with get user

PreafericitulMarcus committed:
fix(#5): Fixed reviews.

chore(#5): Added UserDTO and UserController. In UserController I added addUser method.

Taveeh commented on backend/src/main/java/clf/integra/backend/controller/UserController.java:
```diff
@@ -0,0 +1,22 @@
+    @PostMapping("/users")
+    public UUID addUser(@RequestBody UserDTO user) {
+        System.out.println(user.firstName());
+        return userService.addUserByName(user.firstName(), user.middleName(), user.lastName());
```
```suggestion
    public UUID addUser(@RequestBody UserDTO user) {
        return userService.addUserByName(user.firstName(), user.middleName(), user.lastName());
```

Taveeh commented on backend/src/main/java/clf/integra/backend/service/UserService.java:
```diff
@@ -1,12 +1,25 @@
-
     private final UserRepository userRepository;
+
+    public UUID addUserByName(String firstName, String middleName, String lastName) {
```
```suggestion
    public UUID addUserWithName(String firstName, String middleName, String lastName) {
```

PreafericitulMarcus committed:
Update backend/src/main/java/clf/integra/backend/controller/UserController.java

Co-authored-by: Octavian Custura <octavian.custura@gmail.com>

PreafericitulMarcus committed:
Update backend/src/main/java/clf/integra/backend/service/UserService.java

Co-authored-by: Octavian Custura <octavian.custura@gmail.com>
